File: objective-c-sample/JSQMessagesViewController_SendBird/GroupChannel/View/GroupChannelListTableViewCell.m
//
//  MessagingChannelListTableViewCell.m
//  JSQMessagesViewController_SendBird
//
//  Created by Jed Gyeong on 4/26/16.
//  Copyright © 2016 SENDBIRD. All rights reserved.
//

#import "GroupChannelListTableViewCell.h"

@interface GroupChannelListTableViewCell()

@property (weak, nonatomic) IBOutlet UIImageView *coverImageView;
@property (weak, nonatomic) IBOutlet UILabel *channelTitleLabel;
@property (weak, nonatomic) IBOutlet UILabel *lastMessageLabel;
@property (weak, nonatomic) IBOutlet UILabel *lastMessageDateLabel;
@property (weak, nonatomic) IBOutlet UILabel *unreadMessageCountLabel;
@property (weak, nonatomic) IBOutlet UILabel *memberCountLabel;

@property (strong) SBDGroupChannel *channel;

@end

@implementation GroupChannelListTableViewCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

+ (UINib *)nib
{
    return [UINib nibWithNibName:NSStringFromClass([self class]) bundle:[NSBundle bundleForClass:[self class]]];
}

+ (NSString *)cellReuseIdentifier
{
    return NSStringFromClass([self class]);
}

- (void)setModel:(SBDGroupChannel *)aChannel
{
    self.channel = aChannel;

    SBDUser *representativeUser = nil;

    NSString *channelTitle = @"";
    NSMutableArray<NSString *> *channelTitleNameArray = [[NSMutableArray alloc] init];
    for (SBDUser *user in self.channel.members) {
        if ([user.userId isEqualToString:[SBDMain getCurrentUser].userId]) {
            continue;
        }
        else {
            [channelTitleNameArray addObject:user.nickname];
            representativeUser = user;
        }
    }
    
    if (representativeUser != nil) {
        [self.coverImageView setImageWithURL:[NSURL URLWithString:representativeUser.profileUrl]];
    }
    else {
        [self.coverImageView setImageWithURL:[NSURL URLWithString:self.channel.coverUrl]];
    }
    
    channelTitle = [channelTitleNameArray componentsJoinedByString:@","];

    [self.channelTitleLabel setText:channelTitle];
    NSDate *date = nil;
    if ([[self.channel lastMessage] isKindOfClass:[SBDUserMessage class]]) {
        [self.lastMessageLabel setText:[(SBDUserMessage *)[self.channel lastMessage] message]];
        date = [NSDate dateWithTimeIntervalSince1970:([[self.channel lastMessage] createdAt] / 1000)];
    }
    else if ([[self.channel lastMessage] isKindOfClass:[SBDFileMessage class]]) {
        [self.lastMessageLabel setText:@"(File)"];
        date = [NSDate dateWithTimeIntervalSince1970:([[self.channel lastMessage] createdAt] / 1000)];
    }
    else {
        [self.lastMessageLabel setText:@""];
        date = [NSDate dateWithTimeIntervalSince1970:([self.channel createdAt] / 1000)];
    }

    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
    [formatter setLocale:[NSLocale currentLocale]];
    [formatter setTimeStyle:NSDateFormatterShortStyle];
    [formatter setDateStyle:NSDateFormatterShortStyle];
    [formatter setTimeZone:[NSTimeZone localTimeZone]];
    
    [self.lastMessageDateLabel setText:[formatter stringFromDate:date]];

    if ([self.channel unreadMessageCount] == 0) {
        [self.unreadMessageCountLabel setHidden:YES];
    }
    else {
        [self.unreadMessageCountLabel setHidden:NO];
        [self.unreadMessageCountLabel setText:[NSString stringWithFormat:@"%lu", self.channel.unreadMessageCount]];
    }
    
    [self.memberCountLabel setText:[NSString stringWithFormat:@"%lu", self.channel.memberCount]];
}

- (void)drawRect:(CGRect)rect {
    self.coverImageView.layer.cornerRadius = 18;
    self.coverImageView.clipsToBounds = YES;
}


@end

File: swift-sample/SampleUI/GroupChannelListTableViewCell.swift
//
//  GroupChannelListTableViewCell.swift
//  SampleUI
//
//  Created by Jed Kyung on 8/24/16.
//  Copyright © 2016 SendBird. All rights reserved.
//

import UIKit
import SendBirdSDK

class GroupChannelListTableViewCell: UITableViewCell {
    var channel: SBDGroupChannel?
    
    @IBOutlet private weak var coverImageView: UIImageView!
    @IBOutlet private weak var channelTitleLabel: UILabel!
    @IBOutlet private weak var lastMessageLabel: UILabel!
    @IBOutlet private weak var lastMessageDateLabel: UILabel!
    @IBOutlet private weak var unreadMessageCountLabel: UILabel!
    @IBOutlet private weak var memberCountLabel: UILabel!

    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

    override func setSelected(selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }
    
    static func nib() -> UINib {
        return UINib.init(nibName: NSStringFromClass(self).componentsSeparatedByString(".").last!, bundle: NSBundle(forClass: self));
    }
    
    static func cellReuseIdentifier() -> String {
        return NSStringFromClass(self).componentsSeparatedByString(".").last!
    }
    
    func setModel(aChannel: SBDGroupChannel) {
        self.channel = aChannel
        
        var representativeUser: SBDUser?
        
        var channelTitle: String?
        var channelTitleNameArray: [String] = []
        for item in self.channel!.members! {
            let user = item as! SBDUser
            if user.userId == SBDMain.getCurrentUser()!.userId {
                continue
            }
            else {
                channelTitleNameArray.append(user.nickname!)
                representativeUser = user
            }
        }
        
        if representativeUser != nil {
            self.coverImageView.af_setImageWithURL(NSURL.init(string: (representativeUser?.profileUrl)!)!)
        }
        else {
            self.coverImageView.af_setImageWithURL(NSURL.init(string: (self.channel?.coverUrl)!)!)
        }
        
        channelTitle = channelTitleNameArray.joinWithSeparator(",")
        
        self.channelTitleLabel.text = channelTitle
        let date: NSDate?
        if self.channel?.lastMessage?.isKindOfClass(SBDUserMessage) == true {
            self.lastMessageLabel.text = (self.channel?.lastMessage as! SBDUserMessage).message
            date = NSDate(timeIntervalSince1970: Double((self.channel?.lastMessage!.createdAt)!) / 1000)
        }
        else if self.channel?.lastMessage?.isKindOfClass(SBDUserMessage) == true {
            self.lastMessageLabel.text = "(File)"
            date = NSDate(timeIntervalSince1970: Double((self.channel?.lastMessage!.createdAt)!) / 1000)
        }
        else {
            self.lastMessageLabel.text = ""
            date = NSDate(timeIntervalSince1970: Double((self.channel?.createdAt)!) / 1000)
        }
        
        let formatter = NSDateFormatter()
        formatter.locale = NSLocale.currentLocale()
        formatter.timeStyle = NSDateFormatterStyle.ShortStyle
        formatter.dateStyle = NSDateFormatterStyle.ShortStyle
        formatter.timeZone = NSTimeZone.localTimeZone()
        
        self.lastMessageDateLabel.text = formatter.stringFromDate(date!)
        
        if self.channel?.unreadMessageCount == 0 {
            self.unreadMessageCountLabel.hidden = true
        }
        else {
            self.unreadMessageCountLabel.hidden = false
            self.unreadMessageCountLabel.text = String(format: "%lu", (self.channel?.unreadMessageCount)!)
        }
        
        self.memberCountLabel.text = String(format: "%lu", self.channel!.memberCount)
    }
    
    override func drawRect(rect: CGRect) {
        self.coverImageView.layer.cornerRadius = 18
        self.coverImageView.clipsToBounds = true
    }
}
